src/main/resources/jupyter/edit-mode.js

```diff
@@ -0,0 +1,337 @@
+                console.error(err)
+                removeElementById(modeBannerId)
+                toggleMetaFailureBanner(true)
+            })
+    }
+
+    //this function assumes any status not included in these lists represents an in sync notebook to defend against future fields being added being auto-categorized as failures
+    function handleCheckMetaResp(res) {
+        meta = res //set meta state
+
+        const outOfSyncStatuses = ["DESYNCHRONIZED", "LOCAL_CHANGED", "REMOTE_CHANGED"]
+        const fatalSyncStatuses = ["REMOTE_NOT_FOUND"]
+
+        if (outOfSyncStatuses.includes(res.syncStatus)) {
+            promptUserWithModal(syncIssueTitle, syncIssueButtons, syncIssueBody)
+        } else if (fatalSyncStatuses.includes(res.syncStatus)) {
+            promptUserWithModal(syncIssueTitle, noRemoteFileButtons, syncIssueFatalBody)
+        } else {
+            console.info('healthy sync status detected')
+        }
+    }
+
+    function maintainLock(res) {
+        const isEditMode = res.syncMode == "EDIT"
+        if (isEditMode) {
+            getLock()
+        }
+
+        renderModeBanner(isEditMode)
+    }
+
+    async function getLock() {
+        const payload = {
+            ...basePayload,
+            method: 'POST',
+            body: JSON.stringify(getLocalPath())
+        }
+
+        fetch(lockUrl, payload)
+            .then(res => {
+                handleLockStatus(res)
+                return res.json()
+            })
+            .catch(err => {
+                console.error(err)
+            })
+    }
+
+    async function toggleMetaFailureBanner(shouldShow) {
+        const bannerId = "notification_metaFailure"
+        const bannerText = "Failed to check notebook status, changes may not be saved to workspace. Retrying..."
+
+        removeElementById(bannerId)
+
+        if (shouldShow) {
+            const bannerStyling = "btn btn-xs navbar-btn btn-danger"
+
+            $('#notification_area').append(
+                $('<div>').attr({
+                    "id": bannerId,
+                    "class": bannerStyling
+                })
+                .append($('<span>').html('<i class="fa fa-exclamation-triangle"></i> ' + bannerText))
+            );
+        }
+
+    }
+
+    async function handleLockStatus(res) {
+        if (!res.ok) {
+            const status = res.status
+            const errorText = res.statusText
+
+            if (status == 409) {
+                res.json().then(body => {
+                    const message = getLockConflictBody(body.lockedBy)
+                    promptUserWithModal(lockConflictTitle, lockIssueButtons, message)
+                })
+            }
+            //for the lock endpoint, we consider all non 'ok' statuses an error
+            throw new Error(errorText)
+        }
+    }
+
+    const getLockConflictBody = (lockHolder) => {
+        return `<p>This file is currently being editted by ${lockHolder}.</p>` +
```

Comment:
Unfortunately we won't have the specifics here (per the user metadata privacy doc) - will just have to say "by another user"